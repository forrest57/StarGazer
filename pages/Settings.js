import { StatusBar } from 'expo-status-bar'
import { View, SafeAreaView, FlatList, Text, StyleSheet } from 'react-native'
import tw from 'twrnc'
import { Ionicons } from '@expo/vector-icons'
import { SimpleLineIcons } from '@expo/vector-icons'
import AnimatedButton from '../components/animatedButton'

const chevronLeftIcon = () => (
  <View
    style={tw` bg-gray-700 flex items-center justify-center rounded-md p-4 w-full`}>
    <Ionicons name='chevron-back-sharp' size={25} color='gray' />
  </View>
)
const gearIcon = () => (
  <View
    style={tw` bg-gray-700 flex items-center justify-center rounded-md p-4 w-full `}>
    <SimpleLineIcons name='settings' size={25} color='gray' />
  </View>
)

const ResetButtonText = () => (
  <View
    style={tw` bg-gray-700 flex items-center justify-center rounded-md p-4 w-3/4`}>
    <Text style={styles.buttonText}>Reset app data</Text>
  </View>
)

export default SettingsPage = ({ navigation, route }) => {
  const goBack = () => navigation.goBack(null)

  return (
    <SafeAreaView
      style={[
        { marginTop: StatusBar.currentHeight || 0 },
        tw`flex flex-col bg-gray-800 pt-7 h-full w-full `,
      ]}>
      <View style={tw`flex items-center justify-between flex-row w-3/5 ml-4`}>
        <AnimatedButton Component={chevronLeftIcon} pressFunction={goBack} />
        <Text style={styles.header}>Options</Text>
      </View>
      <View style={tw`h-full items-center justify-center`}>
        <View
          style={tw`flex flex-row 
          bg-gray-700 w-3/4 h-10
          items-center p-1 
          justify-center  px-5 mb-3 rounded-md`}></View>
        <AnimatedButton width='full' Component={ResetButtonText} />
      </View>
    </SafeAreaView>
  )
}

const styles = StyleSheet.create({
  header: {
    marginLeft: 8,
    color: 'gray',
    fontSize: 28,
  },
  buttonText: {
    color: 'gray',
    fontSize: 28,
    fontWeight: 'bold',
  },
})
