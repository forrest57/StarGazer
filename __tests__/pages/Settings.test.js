
import React from 'react'
import { TextInput } from 'react-native'
import renderer from 'react-test-renderer'
// import App from '../App'
import Settings from '../../pages/Settings'

describe('<Settings />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Settings />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  
})
