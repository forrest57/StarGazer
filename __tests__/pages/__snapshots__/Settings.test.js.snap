// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "marginTop": 0,
      },
      Object {
        "backgroundColor": "#1f2937",
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "paddingTop": 28,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginLeft": 16,
        "width": "60%",
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "display": "flex",
          "marginTop": 8,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
          "width": "auto",
        }
      }
    >
      <View
        accessible={true}
        activeOpacity={1}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 6,
            "display": "flex",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#374151",
              "borderRadius": 6,
              "display": "flex",
              "justifyContent": "center",
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
              "width": "100%",
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "gray",
          "fontSize": 28,
          "marginLeft": 8,
        }
      }
    >
      Options
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#374151",
          "borderRadius": 6,
          "display": "flex",
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 12,
          "paddingBottom": 4,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 4,
          "width": "75%",
        }
      }
    />
    <View
      collapsable={false}
      style={
        Object {
          "display": "flex",
          "marginTop": 8,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        activeOpacity={1}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": 6,
            "display": "flex",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#374151",
              "borderRadius": 6,
              "display": "flex",
              "justifyContent": "center",
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
              "width": "75%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "gray",
                "fontSize": 28,
                "fontWeight": "bold",
              }
            }
          >
            Reset app data
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
